project_euler/problem_089/sol1.py

```diff
@@ -0,0 +1,141 @@
+"""
+Project Euler Problem 89: https://projecteuler.net/problem=89
+
+For a number written in Roman numerals to be considered valid there are basic rules
+which must be followed. Even though the rules allow some numbers to be expressed in
+more than one way there is always a "best" way of writing a particular number.
+
+For example, it would appear that there are at least six ways of writing the number
+sixteen:
+
+IIIIIIIIIIIIIIII
+VIIIIIIIIIII
+VVIIIIII
+XIIIIII
+VVVI
+XVI
+
+However, according to the rules only XIIIIII and XVI are valid, and the last example
+is considered to be the most efficient, as it uses the least number of numerals.
+
+The 11K text file, roman.txt (right click and 'Save Link/Target As...'), contains one
+thousand numbers written in valid, but not necessarily minimal, Roman numerals; see
+About... Roman Numerals for the definitive rules for this problem.
+
+Find the number of characters saved by writing each of these in their minimal form.
+
+Note: You can assume that all the Roman numerals in the file contain no more than four
+consecutive identical units.
+"""
+
+import os
+
+symbols = {"I": 1, "V": 5, "X": 10, "L": 50, "C": 100, "D": 500, "M": 1000}
+
+
+def parse_roman_numerals(numerals: str) -> int:
+    """
+    Converts a string of roman numerals to an integer.
+    e.g.
+    >>> parse_roman_numerals("LXXXIX")
+    89
+    >>> parse_roman_numerals("IIII")
+    4
+    """
+
+    num = 0
+
+    i = 0
```

Comment:
Descriptive names please.

Reply:
It's a looping index counter, i is pretty much universally used. Not sure what you're expecting in its place.